homeassistant/components/aten_pe/switch.py

```diff
@@ -0,0 +1,121 @@
+"""The ATEN PE switch component."""
+
+import logging
+import voluptuous as vol
+
+from homeassistant.components.switch import (
+    DEVICE_CLASS_OUTLET,
+    SwitchDevice,
+    PLATFORM_SCHEMA,
+)
+from homeassistant.const import CONF_HOST, CONF_PORT, CONF_USERNAME
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_COMMUNITY = "community"
+CONF_AUTHKEY = "authkey"
+CONF_PRIVKEY = "privkey"
+
+DEFAULT_HOST = "localhost"
+DEFAULT_PORT = "161"
+DEFAULT_COMMUNITY = "private"
+DEFAULT_USERNAME = "administrator"
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
+    {
+        vol.Optional(CONF_HOST, default=DEFAULT_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+        vol.Optional(CONF_COMMUNITY, default=DEFAULT_COMMUNITY): cv.string,
+        vol.Optional(CONF_USERNAME, default=DEFAULT_USERNAME): cv.string,
+        vol.Required(CONF_AUTHKEY): cv.string,
+        vol.Required(CONF_PRIVKEY): cv.string,
+    }
+)
+
+
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Set up the ATEN PE switch."""
+    from atenpdu import AtenPE
+
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    community = config.get(CONF_COMMUNITY)
+    username = config.get(CONF_USERNAME)
+    authkey = config.get(CONF_AUTHKEY)
+    privkey = config.get(CONF_PRIVKEY)
+
+    dev = AtenPE(
+        node=host,
+        serv=port,
+        community=community,
+        username=username,
+        authkey=authkey,
```

Comment:
Why not directly use the config options? Instead of making vars:

```suggestion
        authkey=config[CONF_AUTHKEY],
```